motoman_ar2010_support/urdf/ar2010_macro.xacro

```diff
@@ -112,7 +112,7 @@
       <parent link="${prefix}link_1_s"/>
       <child link="${prefix}link_2_l"/>
       <origin xyz="${0.150 + kinematic_offsets['2_l']['x']} ${0 + kinematic_offsets['2_l']['y']} ${0 + kinematic_offsets['2_l']['z']}"
-              rpy="${radians(90) + kinematic_offsets['2_l']['roll']} ${radians(-90) + kinematic_offsets['2_l']['pitch']} ${radians(-180) + kinematic_offsets['2_l']['yaw']}"/>
+              rpy="${0 + kinematic_offsets['2_l']['roll']} ${radians(-90) + kinematic_offsets['2_l']['pitch']} ${radians(-90) + kinematic_offsets['2_l']['yaw']}"/>
```

Comment:
@rachit-aggarwal , do both the previous and the updated nominal rpy values result in the same transformation matrix?

Also, from our conversation, I get the impression that your updated values match what @milad-khaledyan 's code uses when calculating offsets. And, since we are using the offsets calculated there, we need to use the same nominal values in both places?

@Doomerdinger , where are you getting these rpy values, or how are you calculating them? While we are still using the nominal + offset convention, we have to make sure we are consistent between the URDF and the code that calculates the offsets, as rpy values are not unique and we could end up using different values (like what happened here).

Reply:
@kleinma Yes, the transformation using the nominal (without offsets) has no trouble matching customDH FK for both previous and new values. The issue arises because the euler calculations used by @milad-khaledyan's code yields a different nominal leading to a mismatch in nominal+offset. So this fix uses the nominal from @milad-khaledyan's code.